client/collector/src/main/java/org/hawkular/apm/client/collector/DefaultTraceCollector.java

```diff
@@ -1494,6 +1494,22 @@ public boolean isActive() {
     }
 
     /* (non-Javadoc)
+     * @see org.hawkular.apm.client.api.SessionManager#deactivate()
+     */
+    @Override
+    public void deactivate() {
+        try {
+            if (fragmentManager.hasFragmentBuilder()) {
+                fragmentManager.clear();
+            }
+        } catch (Throwable t) {
```

Comment:
No. This is truly a "catch everything and don't dare impact the application" catch block.